monkey/infection_monkey/monkey.py

```diff
@@ -180,14 +181,19 @@ def _setup(self):
         control_channel.register_agent(self._opts.parent)
 
         config = control_channel.get_config()
-        self._monkey_inbound_tunnel = self._control_client.create_control_tunnel(
-            config.keep_tunnel_open_time
+
+        local_port = get_free_tcp_port()
+        sock, ip_str, port = connect(self._opts.servers)
+        sock.close()
+        self._relay = TCPRelay(
+            local_port,
+            IPv4Address(ip_str),
+            port,
+            client_disconnect_timeout=config.keep_tunnel_open_time,
         )
-        if self._monkey_inbound_tunnel and maximum_depth_reached(
-            config.propagation.maximum_depth, self._current_depth
-        ):
-            self._inbound_tunnel_opened = True
-            self._monkey_inbound_tunnel.start()
+
+        if maximum_depth_reached(config.propagation.maximum_depth, self._current_depth):
+            self._relay.start()
```

Comment:
I see that this logic preserves what was here, but I'm confused. This seems backwards. If maximum depth is reached, we _don't_ want to start the relay; since we won't spawn any new/child agents, there's no need for us to start a relay.

How did this ever work in the first place?

Reply:
Yeah, I was wondering the same thing. The logic was backwards. I pushed commit to fix this.

Reply:
Are you sure it got pushed?

Reply:
It's pushed now